frontend/system/src/Admin/Users/ListView/UserAccountEditModal/UserAccountEditModal.tsx

```diff
@@ -0,0 +1,191 @@
+  isValidUsername,
+  User,
+} from '@uc-frontend/common';
+import { UserAccountRowFragment, useUserAccount } from '../../api';
+import { userAccountParsers } from '../../api/api';
+import { UserAccountEditForm } from './UserAccountEditForm';
+
+interface UserAccountEditModalProps {
+  mode: ModalMode | null;
+  isOpen: boolean;
+  onClose: () => void;
+  userAccount: UserAccountRowFragment | null | User;
+}
+
+const createUserAccountRowFragment = (
+  seed?: UserAccountRowFragment | User | null
+): UserAccountRowFragment => ({
+  __typename: 'UserAccountNode',
+  id: FnUtils.generateUUID(),
+  username: '',
+  displayName: '',
+  permissions: [],
+  auditLogs: [],
+  ...seed,
+});
+
+interface UseDraftUserAccountControl {
+  draft: UserAccountRowFragment;
+  password?: string;
+  setPassword: (password: string) => void;
+  onUpdate: (patch: Partial<UserAccountRowFragment>) => void;
+  onChangeUserAccount: () => void;
+  onSave: () => Promise<void>;
+  isLoading: boolean;
+}
+
+export const useDraftUserAccount = (
+  seed: UserAccountRowFragment | User | null,
+  mode: ModalMode | null
+): UseDraftUserAccountControl => {
+  const [userAccount, setUserAccount] = useState<UserAccountRowFragment>(() =>
+    createUserAccountRowFragment(seed)
+  );
+  const [password, setPassword] = useState('');
+  const nextUserAccount = useUserAccount.document.next(userAccount);
+  const { mutateAsync: insert, isLoading: insertIsLoading } =
+    useUserAccount.document.insert();
+  const { mutateAsync: update, isLoading: updateIsLoading } =
+    useUserAccount.document.update();
+
+  const onUpdate = (patch: Partial<UserAccountRowFragment>) => {
+    setUserAccount({ ...userAccount, ...patch });
+  };
+
+  const onSave = async () => {
+    if (mode === ModalMode.Create) {
+      await insert(userAccountParsers.toCreate(userAccount, password));
+    } else {
+      await update(userAccountParsers.toUpdate(userAccount, password));
+    }
+  };
+
+  const onChangeUserAccount = () => {
+    if (mode === ModalMode.Create) {
+      setUserAccount(createUserAccountRowFragment());
+    } else {
+      setUserAccount(createUserAccountRowFragment(nextUserAccount));
+    }
+  };
+
+  return {
+    draft: userAccount,
+    password,
+    setPassword,
+    onUpdate,
+    onChangeUserAccount,
+    onSave,
+    isLoading: updateIsLoading || insertIsLoading,
+  };
+};
+
+export const UserAccountEditModal: FC<UserAccountEditModalProps> = ({
+  mode,
+  isOpen,
+  onClose,
+  userAccount,
+}) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const { Modal } = useDialog({ isOpen, onClose });
+  const t = useTranslation(['system']);
+  const { draft, password, setPassword, onUpdate, onSave, isLoading } =
+    useDraftUserAccount(userAccount, mode);
+
+  const isInvalid =
+    (!isValidUsername(draft.username) ||
+      (mode === ModalMode.Create && !password) ||
+      draft.permissions.length === 0) ??
+    true;
+
+  // TODO: do we want the logs tab here too?
```

Comment:
I think an audit log is probably going to be useful, but I'd like to do it more like the events in notify, so it can be searched in one place, and we don't need to do loaders on all our user account objects?

Reply:
Yeah nice, I did like that in Notify, that should work well :)